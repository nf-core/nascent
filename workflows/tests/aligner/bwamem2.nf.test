nextflow_pipeline {

    name "bwamem2"
    script "../../../main.nf"

    test("Should run with defaults") {

        when {
            params {
                outdir = "$outputDir"
                aligner = "bwamem2"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml")).match("software_versions") },
                { assert snapshot(
                    workflow.trace.tasks().size(),
                    UTILS.getAllFilesFromDir("$outputDir/preprocessing/fastp/", ".json"),
                    bam("$outputDir/bwamem2/cd4_REP1.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/bwamem2/cd4_REP2.sorted.bam").getSamLinesMD5(),
                    bam("$outputDir/bwamem2/jurkat.sorted.bam").getSamLinesMD5(),
                    path("$outputDir/bwamem2/samtools_stats").list(),
                    path("$outputDir/quality_control/bbsplit").list(),
                    UTILS.getAllFilesFromDir("$outputDir/quality_control/preseq", ".txt"),
                    path("$outputDir/quality_control/rseqc/read_duplication/xls").list(),
                    path("$outputDir/quality_control/rseqc/read_duplication/rscript").list(),
                    path("$outputDir/quality_control/rseqc/infer_experiment").list(),
                    path("$outputDir/quality_control/rseqc/read_distribution").list(),
                    path("$outputDir/coverage_graphs/").list(),
                    path("$outputDir/transcript_identification/homer/cd4.bed"),
                    path("$outputDir/transcript_identification/homer/jurkat.bed"),
                    path("$outputDir/transcript_identification/intersect/").list(),
                    path("$outputDir/transcript_identification/filtered/").list(),
                    path("$outputDir/multiqc/multiqc_report.html").exists(),
                ).match("output_files")
                }
            )
        }
    }
}
