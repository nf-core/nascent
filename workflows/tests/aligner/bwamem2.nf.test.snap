{
    "output_files": {
        "content": [
            90,
            [
                "cd4_REP1.trimmed.fastp.json:md5,bd9a3344c1591d6be4d524451f9dca53",
                "cd4_REP2.trimmed.fastp.json:md5,bc6e3f9ff7835f220535cc393b8eb25f",
                "jurkat.trimmed.fastp.json:md5,65f7247a87479c7157c7357d575336e2"
            ],
            "c529a16c839e85e119b98354f109352d",
            "9e63b682af88fa902cf92b5c485845b1",
            "c2adf5327ff6d4edda2fdad00c7cb9bf",
            [
                "cd4_REP1.sorted.bam.flagstat:md5,863e2d506d5cc4239af98a5f31bbc906",
                "cd4_REP1.sorted.bam.idxstats:md5,b1dd8bcbd23c53c21f0e11082d9315f2",
                "cd4_REP1.sorted.bam.stats:md5,1536c80bae78b2062508e1de210f6387",
                "cd4_REP2.sorted.bam.flagstat:md5,0fd86dbf8f799fad49ba471702979bdc",
                "cd4_REP2.sorted.bam.idxstats:md5,53204e4c6a9f68664087e4a8123be46a",
                "cd4_REP2.sorted.bam.stats:md5,561610e53fb676ac83252712dcac30d4",
                "jurkat.sorted.bam.flagstat:md5,fd5f02b0f02a407447b804b1d80f5421",
                "jurkat.sorted.bam.idxstats:md5,c61af0847c1ad76c06a8de2815975b32",
                "jurkat.sorted.bam.stats:md5,898c6b259ef686bf797ef15646faf29d"
            ],
            [
                "cd4_REP1.coverage.hist.txt:md5,7cbb473be8d3b32ff2e52fdf4e5d10d2",
                "cd4_REP1.coverage.stats.txt:md5,f1471b61ac17dba283d80e08450c7e55",
                "cd4_REP2.coverage.hist.txt:md5,8d5258a0882494bc4e3f1aa6aa5ed685",
                "cd4_REP2.coverage.stats.txt:md5,3cdb4473211f9da44166ffa6aaa5b602",
                "jurkat.coverage.hist.txt:md5,71a893d9d1d55fd47399f6d47b628d6e",
                "jurkat.coverage.stats.txt:md5,381c69a30099d82066a959deab1a2569"
            ],
            [
                "cd4_REP1.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP1.lc_extrap.txt:md5,8633f84ccd5cc725db9af4b33edd63b0",
                "cd4_REP2.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "cd4_REP2.lc_extrap.txt:md5,3ad9e4028c3711e6d46831c10ed04200",
                "jurkat.c_curve.txt:md5,cf4743abdd355595d6ec1fb3f38e66e5",
                "jurkat.lc_extrap.txt:md5,84faa937faa88476c942f330c5762cb5"
            ],
            [
                "cd4_REP1.pos.DupRate.xls:md5,a80db2d20096ca839a7847ec5b11bf75",
                "cd4_REP1.seq.DupRate.xls:md5,c34531fd7578c6f62cbad53b96a7feb9",
                "cd4_REP2.pos.DupRate.xls:md5,06200ab67a60bee71fd168de88c15369",
                "cd4_REP2.seq.DupRate.xls:md5,c82f6d687eacabbab045db34647c3254",
                "jurkat.pos.DupRate.xls:md5,0721c91ab7c640b046689095047657f8",
                "jurkat.seq.DupRate.xls:md5,aba941b1bf0e93f99e39bd507d1c02de"
            ],
            [
                "cd4_REP1.DupRate_plot.r:md5,a6f96b5b87a142dca2e09868deb8222b",
                "cd4_REP2.DupRate_plot.r:md5,a0686d22ba07f33a627c1a106d442a03",
                "jurkat.DupRate_plot.r:md5,acb2fdffc578643503503bf0081eb7ae"
            ],
            [
                "cd4_REP1.infer_experiment.txt:md5,2d8f31fe48cfc1db9d4c46ba59c2c7f3",
                "cd4_REP2.infer_experiment.txt:md5,70ac3bf0d0c50d644ab97e1698021e55",
                "jurkat.infer_experiment.txt:md5,3ec36bed252146eb9cdb68db30c1855e"
            ],
            [
                "cd4_REP1.read_distribution.txt:md5,1fcc6afbb63242818d446b877a832c3a",
                "cd4_REP2.read_distribution.txt:md5,c2762d927c1c12d520ebee8160561189",
                "jurkat.read_distribution.txt:md5,6c02a757ed379a6c19f6855e38b5d909"
            ],
            [
                "cd4_REP1.dreg.bedGraph:md5,8948a8fa86d8f6d413b77983189ff56e",
                "cd4_REP1.minus.bedGraph:md5,2a1c34f9d9ef9ff1b9da7874b9e3aaad",
                "cd4_REP1.minus.bigWig:md5,5280319275c98dcce023779fa389884d",
                "cd4_REP1.plus.bedGraph:md5,1509ec3a921e3109c5914e1bcef8cf33",
                "cd4_REP1.plus.bigWig:md5,72ccab3173f2018a22a4b36841247ba2",
                "cd4_REP2.dreg.bedGraph:md5,29f865a5668fae4a52a41589ac2b3179",
                "cd4_REP2.minus.bedGraph:md5,c530bc34fa3ec7ac49e88ff65f9c2f92",
                "cd4_REP2.minus.bigWig:md5,5e748c794e037f441741f7f409c8c5ad",
                "cd4_REP2.plus.bedGraph:md5,a675141da2874ec08d91591e5ea8242b",
                "cd4_REP2.plus.bigWig:md5,08674d52e9eeb08807c33ed3e4b3d504",
                "jurkat.dreg.bedGraph:md5,c25a4fb095e9f7d6766a3ce33e08f7d8",
                "jurkat.minus.bedGraph:md5,8d5d9a41df6eb6c56b1bfd3f39dc1fc6",
                "jurkat.minus.bigWig:md5,d06c8015c996bf520fff17266fd01f84",
                "jurkat.plus.bedGraph:md5,6ed63e5983edaa74fb3965676efdb674",
                "jurkat.plus.bigWig:md5,7a02334f2c7300ffdb5a2253c0937390"
            ],
            "cd4.bed:md5,b55e5290d78941f36c3d1ecfef8e0062",
            "jurkat.bed:md5,383cfaf10535dbe5d7f47607e345f4cb",
            [
                "cd4_intersect.bed:md5,a5d31a685d89293ae64daff9a9054568",
                "jurkat_intersect.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
            ],
            [
                "cd4_filtered.bed:md5,686cb2e55d18a584632711be1874a9f8",
                "jurkat_filtered.bed:md5,f1dde43c4ad9dec972ff9fa38cc6f2fe"
            ],
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T14:30:34.98243"
    },
    "software_versions": {
        "content": [
            {
                "BBMAP_PILEUP": {
                    "bbmap": 39.01,
                    "samtools": "1.16.1",
                    "pigz": 2.6
                },
                "BEDTOOLS_GENOMECOV_MINUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_GENOMECOV_PLUS": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_INTERSECT_FILTER": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_MERGE": {
                    "bedtools": "2.31.1"
                },
                "BEDTOOLS_SORT": {
                    "bedtools": "2.31.1"
                },
                "BWAMEM2_INDEX": {
                    "bwamem2": "2.2.1"
                },
                "BWAMEM2_MEM": {
                    "bwamem2": "2.2.1",
                    "samtools": "1.19.2"
                },
                "CAT_CAT": {
                    "pigz": "2.3.4"
                },
                "CUSTOM_GETCHROMSIZES": {
                    "getchromsizes": 1.2
                },
                "DEEPTOOLS_BAMCOVERAGE_MINUS": {
                    "deeptools": "3.5.1"
                },
                "DEEPTOOLS_BAMCOVERAGE_PLUS": {
                    "deeptools": "3.5.1"
                },
                "FASTP": {
                    "fastp": "0.23.4"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GTF2BED": {
                    "perl": "5.26.2"
                },
                "HOMER_MAKETAGDIRECTORY": {
                    "homer": 4.11,
                    "samtools": 1.11
                },
                "PINTS_CALLER": {
                    "python": "3.12.6",
                    "pints": "1.1.13"
                },
                "PRESEQ_CCURVE": {
                    "preseq": "3.1.1"
                },
                "PRESEQ_LCEXTRAP": {
                    "preseq": "3.1.1"
                },
                "RSEQC_INFEREXPERIMENT": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDISTRIBUTION": {
                    "rseqc": "5.0.2"
                },
                "RSEQC_READDUPLICATION": {
                    "rseqc": "5.0.2"
                },
                "SAMTOOLS_FLAGSTAT": {
                    "samtools": 1.2
                },
                "SAMTOOLS_IDXSTATS": {
                    "samtools": 1.2
                },
                "SAMTOOLS_INDEX": {
                    "samtools": 1.2
                },
                "SAMTOOLS_SORT": {
                    "samtools": 1.2
                },
                "SAMTOOLS_STATS": {
                    "samtools": 1.2
                },
                "SUBREAD_FEATURECOUNTS_GENE": {
                    "subread": "2.0.1"
                },
                "SUBREAD_FEATURECOUNTS_PREDICTED": {
                    "subread": "2.0.1"
                },
                "Workflow": {
                    "nf-core/nascent": "v2.3.0dev"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T14:30:34.70143"
    }
}